src/fixit/rules/pyflakes/no_double_import.py

```diff
@@ -0,0 +1,210 @@
+            self.report(node, message=self._message(double_imports), replacement=new_node)
+
+    def visit_Import(self, node: cst.Import) -> None:
+        self.get_double_import(node)
+
+    def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
+        self.get_double_import(node)
+
+    def leave_Import(self, node: cst.Import) -> None:
+        return self.delete_double_import(node)
+
+    def leave_ImportFrom(self, node: cst.ImportFrom) -> None:
+        return self.delete_double_import(node)
+
+    VALID = [
+        Valid(
+            """
+            import a
+
+            x = a.b([1, 3, 5, 6, 8])
+            """
+        ),
+        Valid(
+            """
+            def fn1():
+                import a
+                return a.b
+
+            def fn2():
+                import a
+                return a.c
+            """
+        ),
+        Valid(
+            # valid, not good
+            """
+            import a as f
+            import a
+            """
+        ),
+        Valid(
+            """
+            import a
+            from a import b
+            """
+        ),
+        Valid(
+            """
+            import a
+            import a.b.c.d
+            """
+        ),
+        Valid(
+            """
+            import a
+            from a import *
+            """
+        ),
+        # This is allowable to cover the case that we have a `from a import fn`
+        # statement in another file 
+        Valid(
+        """
+            import a
+
+            def fn():
+                import a
+                return a.b
+        """
+        )
+    ]
+
+    INVALID = [
+        Invalid(
+            """
+            import a
+            import b
+            import a
+            """,
+            expected_replacement="""
+            import a
+            import b
+            """,
+        ),
+        Invalid(
+            """
+            import g, e as g
+
+            x = g.b([1, 3, 5, 6, 8])
+            """,
+            expected_replacement="""
+            import e as g
+
+            x = g.b([1, 3, 5, 6, 8])
+            """,
+        ),
+        Invalid(
+            """
+            import a, b, a
+            """,
+            expected_replacement="""
+            import a, b
+            """,
+        ),
+        Invalid(
+            """
+            import a, b
+            import a
+            """,
+            expected_replacement="""
+            import a, b
+            """,
+        ),
+        # TODO(@ansley): The next two cases will be autofixed as shown in the
+        # "expected replacement" section, although this isn't the best way to
+        # handle the situation. We should add additional logic for this special
+        # case or create a completely different lint rule.
+        Invalid(
+            """
+            import a
+            from x import a
+            """,
+            expected_replacement="""
+            import a
+            """,
```

Comment:
For these shadowed imports, this autofix is incorrect — it will result in different runtime behavior than what the original code was, because the last import of a name overwrites all previous imports of that name.

If we can't get that right, then we need to skip the autofix, or defer this rule until a future time when we can do it correctly.